### DISCUSSION TITLE
How to obtain latitude and longitude?

### DISCUSSION THREAD

**Discussion by RezaMortezaee** [clickbar/laravel-magellan#108] [Q&A]
Hi,
Can anyone please tell me how I should obtain latitude and longitude to make a point class?

In the example of package documentation, should the Port model have separate latitude and longitude fields in the database with a data type of float plus a location field? 

As an example, is this what it should be like:
'location' => Point::makeGeodetic($this->latitude , $this->longitude)
OR there are other ways to get those?

**Comment by saibotk** [comment]
Hey 👋 

You only need those when you create a Point in code manually. The database can store both coordinates in a single field and magellan casts it to a Point.

Use the magellanPoint() function in a migration to create such a column in the database.

Hope this helps 🙌

**Reply by RezaMortezaee** [reply]
Thank you so much for your reply.

I actually did it (the magellanPoint() function in a migration). 
As mentioned in the package doc, to create a location, this is what it should be like:
`'location' => Point::makeGeodetic(49.87108851299202, 8.625026485851762)` 
BUT the problem/ambiguity for me is:
- How those lat & long are coming from?
- How should I handle this in the controller?
- Is it necessary to have lat & long and location as a $fillable property, or is just location enough, or neither?

Also, by assuming lat & long in $fillable property, these are the properly named getters and setters: 
```
protected function latitude(): Attribute
    { return Attribute::make(

            // Accessor
            get: fn(array $attributes) => $attributes['latitude'],

            // Mutator
            set: fn() => ['latitude' => $this->attributes['latitude']]
        );
    }
```
**In my project, All I need is user should  select his/her address on the map with a pointer, and this will be saved in location column.**

**Reply by studnitz** [reply]
Hi, in a controller you would just write something like.

```php
Something::create([
    // rest of the properties
    'location' => Point::makeGeodetic($request->latitude, $request->longitude),
]);
```

This is actually how we handle a Marker input. No need for extra `Attribute`s for `latitude` and `longitude` and also no need to add them as separate columns. Just make sure you have `latitude` and `longitude` set in the POST/PUT request.
